pkg/server/plugin/upstreamauthority/spire/spire.go

```diff
@@ -29,10 +39,22 @@ func BuiltIn() catalog.Plugin {
 }
 
 type Plugin struct {
-	mtx sync.RWMutex
-
+	mtx         sync.RWMutex
 	trustDomain url.URL
 	config      *Configuration
+
+	nodeMtx    sync.RWMutex
+	nodeClient node.NodeClient
+	conn       *grpc.ClientConn
+	log        hclog.Logger
+
+	pollMtx                sync.RWMutex
```

Comment:
nit: not using the read portion of the lock, change to `sync.Mutex` (more self-documenting)